lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystemFactory.java

```diff
@@ -96,6 +97,9 @@ public static HttpClient createAzureHttpClient(OkHttpClient okHttpClient, HttpCl
         Integer poolSize = clientOptions.getMaximumConnectionPoolSize();
         // By default, OkHttp uses a maximum idle connection count of 5.
         int maximumConnectionPoolSize = (poolSize != null && poolSize > 0) ? poolSize : 5;
+        Dispatcher dispatcher = new Dispatcher();
+        dispatcher.setMaxRequests(Runtime.getRuntime().availableProcessors() * 4);
+        dispatcher.setMaxRequestsPerHost(Runtime.getRuntime().availableProcessors() * 2);
```

Comment:
This is the same as the default for the number of worker threads: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java#L62

I didn't want to get it from that config to avoid adding a dependency on `trino-main`. I set the max requests as twice as much, but I don't have any specific reason for it. I'm open for suggestions how to choose a better value.

Reply:
is this worth exposing it as a configuration property? 

Also nit: extract Runtime.getRuntime().availableProcessors() to a variable, and also inline `dispatcher` below

Reply:
I can't inline it, as the setters don't return anything.

If we'll make it configurable, it should only be a fallback in case we get the default wrong. Maybe we should ask differently - in which cases the number of worker threads should be customized?